src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -74,7 +74,6 @@ public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttp
                         // authentication is requested.
                         try {
                             disconnect();
-                            authenticate();

```

Comment:
This looks weird to me. The next line attempts to resend request after disconnecting, which will not work (how it can work?) Since authenticate is not called anymore, there will be no init and execution.execute will guaranteed to fail, isn't it?

Reply:
Agree. Changed it, so now just letting the failed request go, and the next cycle request will succeed

Reply:
There is another potential issue here. You should only execute the logic here if the request you are intercepting is not coming from authenticate() call itself. Otherwise it will get into infinite loop. Maybe that was the actual cause if the bug. Pretty much you condition in line 64 should be if (response == 403 && request is not "authenticate") { then try to recover }

Reply:
Yes, this is reasonable, thank you. Adjusted the code section